### DISCUSSION TITLE
Test are always run with codeunit Test Runner - Isol. Codeunit

### DISCUSSION THREAD

**Discussion by Bertverbeek4PS** [microsoft/AL-Go#1565] [Ideas]
When the tests are run in the pipelines all tests are kicked of with the codeunit "Test Runner - Isol. Codeunit".
Only if you want to test the API's you must use "Test Runner - Isol. Disabled".
So it would be very helpfull that you can define in the pipeline or something else which test are runing with the isolation codeunit and some tests with the isolation disabled.
Otherwise you cannot test your API's because you must commit a value into the database when you do a GET request.

Moslty you run first all the test with the isolation codeunit and after those are finished you have to run the tests with isolation disabled.

**Comment by freddydk** [comment]
One solution could be to add a testsettings.json file in the same location as the test app (same location you would place disabledtests.json) and in that you could specify settings for tests that are not available in settings.
TestCodeunit could be one.

**Comment by freddydk** [comment]
Will move this to the backlog as a suggestion.
I will work on getting the backlog up-to-date over the next week and have areas to make it easier to combine work items related when working on them

**Comment by SchulzOli** [comment]
for tracking reasons
is this a point in you roadmap? 
maybe
this https://github.com/orgs/microsoft/projects/521/views/2?pane=issue&itemId=44385805
or this https://github.com/orgs/microsoft/projects/521/views/2?pane=issue&itemId=12835584
?

thanks in advance

**Comment by Drakonian** [comment]
Do you accept pull requests for implementing features like this?

Are there any updates on this issue? It's surprising that it's remained unresolved for so many years.

**Reply by Drakonian** [reply]
I added PR

https://github.com/microsoft/AL-Go/pull/2216
